### ISSUE TITLE
skipCommit is polluting master tags

### ISSUE THREAD

**Issue by rajput2107** [jscutlery/semver#581]
Hi,
First of all thanks for the amazing feature skip commit which i needed a lot.

My use case is to skip all the versioning commits in dev branches (checked out from master for development), after skipCommit in dev branch, if only affected libraries are built it works fine and add the tag to the last commit of dev branch only.

But in case we create versions for the unaffected libraries too(using --allowEmptyRelease), let's say by running --skip-nx-cache which will trigger all the libraries, it goes and attach the pre release tags in master  to that specific library old tag commit.

For ex:- 
```
LibraryX - v1.0.0
LibraryY - v1.0.0
LibraryZ - v1.0.0
```

These are in master, I checked out a branch from master and made change only in LibraryX, but I want to bump all the lib versions. 
Then it will attach `LibraryX-1.0.1-dev` into the dev branch last commit but `LibraryY-1.0.1-dev` and `LibraryZ-1.0.1-dev` dev versions are directly attached into the master (on LibraryY-1.0.0 commit) which will pollute the master tags.

Now my master history looks like this even if **_dev branch is not yet merged_** to master.
```
LibraryX - v1.0.0
LibraryY - v1.0.0, LibraryY-1.0.1-dev
LibraryZ - v1.0.0, LibraryZ-1.0.1-dev
```

**Comment by cakeinpanic** [comment]
@rajput2017 which behavior you'd prefer? not to put tags on commits which already exist in master?

**Comment by rajput2107** [comment]
I'm using master latest tags to create production builds (Checking the affected libraries between latest tag in master with master HEAD).
So, if development branches start putting tags to master it'll be chaotic(for me atleast xD).
In my opinion we shouldn't modify other branch commits from some other branch.

**Comment by cakeinpanic** [comment]
@rajput2107 but no commits are modified, it's the beauty of tags

**Comment by rajput2107** [comment]
@cakeinpanic Yes, commits are unaffected; my phrase was incorrect. The only issue is that the master git tags history begins to look bizarre.
Additionally, I just wanted to know why we can't include other tags on the last commit of that branch too?

**Comment by cakeinpanic** [comment]
@rajput2107 sorry, I did not get your question `why we can't include other tags on the last commit of that branch too`, can you elaborate?

**Comment by rajput2107** [comment]
Let's say I checked out a branch from master -> feat1
So, can't we keep all the tags generated in feat1 limited to feat1 only.
Instead of feat1 pushing tags to the old commits of master.

**Comment by edbzn** [comment]
Dropping one important note, tags are linked to commits, not branches. That said I'm not sure to understand your issue.
